src/test/scala/com/cognite/sdk/scala/common/OAuth2SessionTest.scala

```diff
@@ -131,11 +131,32 @@ class OAuth2SessionTest extends AnyFlatSpec with Matchers with OptionValues {
 
     an[SdkException] shouldBe thrownBy {
       OAuth2
-        .SessionProvider[IO](session)
+        .SessionProvider[IO](session, tokenProvider = MockK8sServiceTokenProvider())
+        .unsafeRunTimed(1.second)
+        .value
+        .getAuth
+        .unsafeRunSync()
+    }
+  }
+
+  it should "throw an sdk exception when failing to get k8s service token" in {
+    val session = OAuth2.Session(
+      "https://bluefield.cognitedata.com",
+      123,
+      "sessionKey-value",
+      "irrelevant"
+    )
+
+    implicit val mockSttpBackend: SttpBackendStub[IO, Any] = SttpBackendStub(implicitly[MonadError[IO]])
+
+    val sdkException = the[SdkException ] thrownBy {
+      OAuth2
+        .SessionProvider[IO](session, tokenProvider = MockK8sServiceTokenProvider(false))
```

Comment:
This would become `IO.raiseError(new RuntimeException("Could not get Kubernetes JWT"))`

Reply:
done